skimage/transform/_geometric.py

```
@@ -538,46 +619,44 @@ class EssentialMatrixTransform(FundamentalMatrixTransform):
 
     """
 
+    # Threshold for determinant of rotation matrix.
+    _rot_det_tol = 1e-6
+
+    # Threshold for difference of translation vector from unit length.
+    _trans_len_tol = 1e-6
+
     def __init__(
-        self, rotation=None, translation=None, matrix=None, *, dimensionality=2
+        self, *, rotation=None, translation=None, matrix=None, dimensionality=None
```

Making most or all arguments keyword only makes sense, but I think we can't do that yet without a deprecation cycle? As it is now, this would break users without warning? Or am I missing something?

---

> Making most or all arguments keyword only makes sense, but I think we can't do that yet without a deprecation cycle? As it is now, this would break users without warning? Or am I missing something?

Did you catch the discussion up at https://github.com/scikit-image/scikit-image/pull/7754#issuecomment-2726479573 ?

In more detail, I found one valid hit for Github search `/(?-i)Affine(?-i)Transform\(None/ AND (path:*.py OR path:*.ipynb)` - see below.   Nothing relevant for `/(?-i)Euclidean(?-i)Transform\(\s*None/ AND (path:*.py OR path:*.ipynb)` or `/(?-i)Similarity(?-i)Transform\(\s*None/ AND (path:*.py OR path:*.ipynb)` or `/(?-i)EssentialMatrix(?-i)Transform\(\s*None/ AND (path:*.py OR path:*.ipynb)`.  

I concluded that it was reasonably safe to go straight to keyword only - we would expect very few users to get a new error.  I did a PR to the only person I could find with Github code that would be affected (see above).

---

I think that's reasonable due diligence @lagru @matthew-brett. It's also important to note that it wouldn't just silently pass with different results, but rather, it would fail with a very clear path forward for users.

Having said that, the previous case *could* be caught by checking for length of a `*args` and processing them accordingly, and emitting a deprecation warning. I'll let @stefanv chime in on whether he thinks it's worth the trouble...

---

Personally, in light of https://github.com/scikit-image/scikit-image/pull/7754#issuecomment-2726479573, I wouldn't require a deprecation cycle:

* because in practice it's reasonable (it won't just silently pass and give different results, as @jni pointed out);
* by principle because this change is a *fix* (it's not just any API change).

---

(To add, I don't mean to block this if people agree it's not necessary :wink:)

---

> I also wouldn't categorize the mismatch between argument order and application order a bug.

Admittedly, it's an inconsistency that's potentially confusing and/or misleading.